jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java

```diff
@@ -738,6 +851,14 @@ public void abortWorkItem(WorkItem workItem,
 
     protected HttpClient getHttpClient(Integer readTimeout,
                                        Integer connectTimeout) {
+        if (getDoCacheClient() && HTTP_CLIENT_API_43) {
+            if (cachedClient != null) {
+                return cachedClient;
+            } else {
+                getNewPooledHttpClient(readTimeout,
```

Comment:
shouldn't this return the newly created client? otherwise it will be overridden down the road of this method, isn't it?